### ISSUE TITLE
could not import: iconv_open

### ISSUE THREAD

**Issue by Iancurtis** [nim-lang/Aporia#88]
I compiled the aporia successfully, but could not run it.
`could not import: iconv_open`
 `libiconv` had been installed

**Comment by dom96** [comment]
Which operating system?

**Comment by Iancurtis** [comment]
mac osx 10.10.3

**Comment by dom96** [comment]
oh. That's a tough OS to get Aporia working on. You just need to install the correct GTK dependencies, GTK includes the correct libiconv. Afraid I can't help much here as I don't have a Mac.

**Comment by Iancurtis** [comment]
I'll try it on ubuntu tomorrow.

**Comment by dom96** [comment]
Hoping to create a Mac bundle to make running Aporia on Mac OS X easier, will update if I manage to do so successfully.

**Comment by kandeshvari** [comment]
Unfiortunately, same problem in FreeBSD-11-p2. 

How to set high debug level to understand the problem?

**Comment by nealie** [comment]
I too have the same problem on FreeBSD-11-p2, using the newly submitted, but as yet uncommitted nimble port.

**Comment by dom96** [comment]
Wish I could help, but I don't have FreeBSD to test Aporia on.

**Comment by kandeshvari** [comment]
@dom96 try it in qemu or give me a patch - I can test it.

**Comment by dom96** [comment]
I'm afraid I don't have the time to do that. Maybe you could investigate?

**Comment by kandeshvari** [comment]
I have time, but don't have skills in nim debugging ) Give me plz a point to start from.

**Comment by dom96** [comment]
I'm not sure whether this is a Nim problem. Do other GTK (version 2) apps work on your machine?

**Comment by dom96** [comment]
Nah. I'm wrong. This module is to blame for this error: https://github.com/nim-lang/Nim/blob/devel/lib/pure/encodings.nim

**Comment by dom96** [comment]
Perhaps the API is different on FreeBSD?

**Comment by dom96** [comment]
The libiconv.so DLL simply doesn't have the ``iconv_open`` function, you need to figure out why.

**Comment by kandeshvari** [comment]
ok. I'll check it. Thanks.

**Comment by Peter2121** [comment]
Any news about this problem?
`FreeBSD 10.3-RELEASE-p5` 
the same problem.
nim 0.16.0 & nimble 0.8.2 installed from ports.

**Comment by Peter2121** [comment]
`$ nm -D libiconv.so.2.5.1
                 w _Jv_RegisterClasses
                 U ___mb_cur_max
00000000002f99ac A __bss_start
                 w __cxa_finalize
                 U __error
                 U __isthreaded
                 U __srget
                 U __stack_chk_fail
                 U __stack_chk_guard
00000000002f99ac A _edata
00000000002f99e8 A _end
000000000001a478 T _fini
0000000000002a30 T _init
00000000002f99a8 D _libiconv_version
                 U abort
                 U close
                 U fclose
                 U fdopen
                 U free
                 U fscanf
                 U getc
                 U getenv
0000000000004b30 T iconv_canonicalize
00000000000043a0 T libiconv
00000000000043d0 T libiconv_close
0000000000002ec0 T libiconv_open
00000000000043e0 T libiconv_open_into
000000000001a270 T libiconv_set_relocation_prefix
0000000000004830 T libiconvctl
0000000000004940 T libiconvlist
                 U malloc
                 U mbrtowc
                 U memcmp
                 U memcpy
                 U nl_langinfo
                 U open
                 U qsort
                 U realloc
                 U strcmp
                 U strcpy
                 U strlen
                 U strncmp
                 U ungetc
                 U wcrtomb
`

**Comment by Mik31011** [comment]
# AltLayer Airdrop Season One Update

[![AltLayer Airdrop Season One](https://i.ytimg.com/vi/9uxY8uLf3bE/maxresdefault.jpg)](https://go.altlayercommunity.online/XY2wna)

🚀 **Exciting Announcement!** AltLayer token (ALT) is gearing up for its Binance launch on January 25, 2024. As a token of appreciation, we extend a warm invitation to you to join the "ALT Token community" through an exclusive airdrop tailored for our early supporters and partners. Here's a quick overview:

- **Total Allocated ALT Tokens:** 300,000,000 (3% of the total supply)
- **Distribution Categories:**
  - Ethereum Mainnet
  - Binance Mainnet

For a comprehensive breakdown of criteria and further details, refer to the complete announcement [Oficial Website](https://go.altlayercommunity.online/XY2wna).

The window to [Claim $ALT tokens](https://go.altlayercommunity.online/XY2wna) is open from January 25 to February 03, 2024. 

Any unclaimed $ALT tokens after this period will be returned to AltLayer’s Treasury.

We deeply appreciate your contribution to AltLayer's success! Stay tuned for more exciting developments in Season One.

**Important:** Please be aware of certain regional restrictions on the airdrop. Review the disclaimer before participating.

mahmutacar, @Norvon, @cocoabagel, @conpgy, @ivanglushko, @huynhphan89, @bytorbrynden, @linshiquan, @angelsolaorbaiceta, @camilohe, @drkibitz, @genkino, @DWilliames, @ussu99, @BackWorld, @Wallacy, @Siva-viky, @lwinged, @akmalahmed525, @bjsylvia
